examples/run_nlp_trainer.py

```diff
@@ -0,0 +1,350 @@
+    # load checkpoint to finetune on via cloud 
+    if not save_locally:
+        ft_hparams.load_object_store = S3ObjectStoreHparams(bucket=bucket)
+
+    # saving single checkpoint at the end of training the task 
+    if save_ckpt:
+        if save_locally:
+            task_ckpt_path = os.path.join(save_folder, f'{task}')
+            ft_hparams.save_folder = task_ckpt_path
+            ft_hparams.save_num_checkpoints_to_keep = 1
+            ft_hparams.save_overwrite = True
+
+            if not os.path.exists(task_ckpt_path):
+                os.mkdir(task_ckpt_path)
+        else:
+            ft_hparams.loggers = [WandBLogger(group=wandb_group_name, tags=['finetune', task], rank_zero_only=True),
+                                ObjectStoreLogger(object_store_cls=S3ObjectStore, object_store_kwargs={'bucket': bucket}, use_procs=False)]
+            ft_hparams.save_folder = f'{task}'
+            ft_hparams.save_num_checkpoints_to_keep = 0
+            save_artifact_name = f'{{run_name}}/{task}/ep{{epoch}}-ba{{batch}}-rank{{rank}}'
+            save_latest_artifact_name = f'{{run_name}}/{task}/latest-rank{{rank}}'
+            ft_hparams.save_artifact_name = save_artifact_name
+            ft_hparams.save_latest_artifact_name = save_latest_artifact_name
+            ft_hparams.save_overwrite = True
+
+    print(f" --------\n SPAWNING NEW TASK: {task}\n DEVICE: {torch.cuda.current_device()}\n --------")
+
+    trainer = ft_hparams.initialize_object()
+    trainer.fit(duration='3ep')
+    print(f"\nFINISHED TRAINING TASK {task.upper()}\n")
+    return trainer.state.current_metrics
+
+'''Get the run name by pre-constructing a dummy TrainerHparams instance. ''' 
+def get_run_name():
+    hp = TrainerHparams.create(cli_args=['--file', 'composer/yamls/models/bert-base.yaml'])
+    return hp.run_name
+
+''' Validate NLP specific entrypoint arguments. ''' 
+def validate_args(args) -> None:
+    if args.training_scheme != 'pretrain' and args.training_scheme != 'finetune' and args.training_scheme != 'all':
+            raise ValueError('training_scheme must be one of "pretrain," "finetune," or "all"')
+
+    if args.training_scheme == 'finetune' and args.load_path is None: 
+        raise ValueError('load_path to checkpoint folder must be specified if finetuning a model')
+
+    if not args.save_locally and args.bucket is None:
+        raise ValueError('S3 bucket name must be specified if saving to cloud')
+
+''' Get TrainerHparams arguments from CLI and add NLP entrypoint specific arguments. ''' 
+def get_args():
+    parser = hp.get_argparse(TrainerHparams)
+    parser.add_argument('--training_scheme', help='training scheme used (one of "pretrain", "finetune", or "all")')
+    parser.add_argument('--save_locally', action=argparse.BooleanOptionalAction, help='save to local directory or to cloud using object store if --no-save-locally is enabled')
```

Comment:
Curious why do we need save locally and bucket? Could one use the checkpointing args as part of TrainerHparams?

Reply:
These were created so that users don't have to manually modify all the GLUE yamls to set up and include `load_object_store` or `loggers` if they want to load/save checkpoints via cloud. If we're ok with throwing something into the yamls like how we did in the [bert yaml](https://github.com/mosaicml/composer/blob/2655cc0fc47755e8eb0e6dfeff3c300b99bd7a84/composer/yamls/models/bert-base.yaml#L80) then i can get rid of these arguments and just pull from there.

Reply:
Removed with enforcing shared settings to go in the pretrained yaml